arangod/Aql/Query.cpp

```diff
@@ -1617,7 +1618,7 @@ futures::Future<Result> finishDBServerParts(Query& query, ErrorCode errorCode) {
         network::sendRequest(pool, "server:" + server, fuerte::RestVerb::Delete,
                              "/_api/aql/finish/" + std::to_string(queryId),
                              body, options)
-            .thenValue([ss, &query](network::Response&& res) mutable -> Result {
+            .thenValue([&query](network::Response&& res) mutable -> Result {
```

Comment:
Is this modification safe?
Now we are only capturing query by reference, but we do not ensure that the query is still valid when the `then` is executed. Previously we have captured a shared-ptr copy of the query state `ss`, and in its `executeLocked()` method also checked if the state was still valid.
This is now all gone, and I fear the change is not safe.

Reply:
Yes, correct. Will fix.